mvc/controllers/actions.rst

```diff
@@ -1,8 +1,63 @@
+
+Controllers support :doc:`dependency injection <dependency-injection>`, and should follow the `Explicit Dependencies Principle <http://deviq.com/explicit-dependencies-principle>`_ and request any dependencies their actions require through their constructor. Controllers provide a logical means of grouping similar actions together, allowing common sets of rules (e.g. routing, caching, authorization) to be applied collectively.
+
+Within the Model-View-Controller pattern, a Controller is responsible for the initial processing of the request and instantiation of the Model. Business decisions should ideally be performed within the Model. The Controller then takes the result of the Model's processing (if any), chooses which View should be returned, and returns that View along with any data that View requires. Learn more: :doc:`../overview`.
+
+.. tip:: The Controller is a `UI level` abstraction. Its responsibility is to ensure incoming request data is valid and to choose which View (or result) should be returned. In well-factored apps it will not include data access or business logic, but instead will delegate to services handling these responsibilities.
+ 
+Defining Actions
+----------------
+Any public method on a controller type is an action. Parameters on actions are bound to request data and validated using :doc:`model binding <../models/model-binding>`. Any action that accepts parameters should verify that the incoming data is valid by checking ``ModelState.IsValid``.
+
+.. warning:: Action methods that accept parameters should always check ``ModelState.IsValid`` as part of their processing.
+
+Action methods should contain logic for mapping an incoming request to a business concern. These business concerns should typically be represented as services that your controller accesses through dependency injection. Actions then map the result of the business action to an application state.
+
+Actions can return an ``IActionResult`` that produces a response. You can either create and return your own ``ActionResult`` directly, or you can use one of the many helper methods available on the Controller base class.
+
+Controller Helper Methods
+#########################
+
+Inheriting from the base ``Controller`` class provides the following helper methods for returning various kinds of responses.
+
+View
+	Returns a View that uses a model to render HTML. Example: ``return View(customer);``
+
+HTTP Status Code
+	Return an HTTP status code. Examples: ``return Ok();`` or ``return BadRequest();``
+
+Formatted Response
+	Return ``Json`` or similar to format an object in a specific manner. Example: ``return Json(customer);``
+
+An object
+	Return an object you want formatted based on the client's request. Learn more about :doc:`../models/formatting`
```

Comment:
This isn't really a helper method. It probably deserves it's own section.

I'd mention something like `CreatedAtRoute` in place of this.